src/ray/core_worker/core_worker.cc

```diff
@@ -784,8 +784,8 @@ void CoreWorker::Exit(
          detail = std::move(detail),
          creation_task_exception_pb_bytes]() {
           rpc::DrainServerCallExecutor();
-          Disconnect(exit_type, detail, creation_task_exception_pb_bytes);
           KillChildProcs();
+          Disconnect(exit_type, detail, creation_task_exception_pb_bytes);
```

Comment:
hmm, why does this order matter?

Reply:
It doesn't matter since raylet has code handling failure. This is only to make the async_wait shorter. Not very important.

Reply:
Can we add a comment to this effect?